x-pack/plugin/data-frame/src/main/java/org/elasticsearch/xpack/dataframe/transforms/DataFrameTransformTask.java

```diff
@@ -763,7 +773,30 @@ protected void onFinish(ActionListener<Void> listener) {
                 logger.debug(
                     "Finished indexing for data frame transform [" + transformTask.getTransformId() + "] checkpoint [" + checkpoint + "]");
                 auditBulkFailures = true;
-                listener.onResponse(null);
+                if (isContinuous()) {
```

Comment:
maybe `onStarted` would be the better place? Imagine frequency is set to 1h, a checkpoint is done, the transform is set back to started (idle) and you do the update call and then the scheduler starts the next checkpoint. You would have to wait another round to retrieve the update.

Reply:
@hendrikmuhs `onStarted` MAY work? My thought here was `onStarted` is called on the initial start as well. Meaning, the config was just recently gathered. We might as well not even attempt to gather the config in the executor if we are going to grab the config on started?

Reply:
I am not sure what users will expect if they have a long frequency. Also, with an exceptionally long frequency, we may want to provide the ability for a `_run_now` type of endpoint.

Reply:
@hendrikmuhs I moved the reload to `onStarted` as you suggested.